src/control/drpc/module_svc_test.go

```diff
@@ -0,0 +1,252 @@
+
+	if service == nil {
+		t.Fatal("service was nil")
+	}
+
+	common.AssertEqual(t, len(service.modules), 0, "expected empty module list")
+}
+
+func TestService_RegisterModule_Single_Success(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)
+
+	service := NewModuleService(log)
+	expectedID := defaultTestModID
+	testMod := newTestModule(expectedID)
+
+	err := service.RegisterModule(testMod)
+
+	if err != nil {
+		t.Fatalf("expected no error, got: %v", err)
+	}
+
+	mod, ok := service.modules[expectedID]
+
+	if !ok {
+		t.Fatalf("module wasn't found under ID %d", expectedID)
+	}
+
+	if diff := cmp.Diff(testMod, mod); diff != "" {
+		t.Fatalf("(-want, +got)\n%s", diff)
+	}
+}
+
+func TestService_RegisterModule_Multiple_Success(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)
+
+	service := NewModuleService(log)
+	expectedIDs := []int32{-1, 2, 255, defaultTestModID}
+	testMods := make([]*mockModule, 0, len(expectedIDs))
+
+	for _, id := range expectedIDs {
+		mod := newTestModule(id)
+		testMods = append(testMods, mod)
+
+		err := service.RegisterModule(mod)
+
+		if err != nil {
+			t.Fatalf("expected no error, got: %v", err)
+		}
+	}
+
+	for i, id := range expectedIDs {
+		mod, ok := service.modules[id]
+
+		if !ok {
+			t.Fatalf("registered module %d wasn't found", id)
+		}
+
+		if diff := cmp.Diff(testMods[i], mod); diff != "" {
+			t.Fatalf("(-want, +got)\n%s", diff)
+		}
+	}
+}
+
+func TestService_RegisterModule_DuplicateID(t *testing.T) {
+	log, buf := logging.NewTestLogger(t.Name())
+	defer common.ShowBufferOnFailure(t, buf)
+
+	service := NewModuleService(log)
+	testMod := newTestModule(15)
+	dupMod := newTestModule(testMod.IDValue)
+
+	err := service.RegisterModule(testMod)
+	if err != nil {
+		t.Fatalf("expected no error for initial registration, got: %v",
+			err)
+	}
+
+	err = service.RegisterModule(dupMod)
+
+	common.CmpErr(t, errors.New("already exists"), err)
+}
+
+func getGarbageBytes() []byte {
+	badBytes := make([]byte, 250)
+	for i := range badBytes {
+		badBytes[i] = byte(i)
+	}
+
+	return badBytes
+}
+
+func getCallBytes(t *testing.T, sequence int64, moduleID int32) []byte {
+	call := &Call{
+		Sequence: sequence,
+		Module:   moduleID,
+	}
+
+	callBytes, err := proto.Marshal(call)
+	if err != nil {
+		t.Fatalf("Got error marshalling test call: %v", err)
+	}
+
+	return callBytes
+}
+
+func getResponse(sequence int64, status Status, body []byte) *Response {
+	return &Response{
+		Sequence: sequence,
+		Status:   status,
+		Body:     body,
+	}
+}
+
+func TestService_ProcessMessage(t *testing.T) {
+	const testSequenceNum int64 = 13
+
+	for name, tc := range map[string]struct {
+		callBytes      []byte
+		handleCallErr  error
+		handleCallResp []byte
+		expectedResp   *Response
+	}{
+		"garbage input bytes": {
+			callBytes:    getGarbageBytes(),
+			expectedResp: getResponse(-1, Status_FAILURE, nil),
+		},
+
```

Comment:
style: no need for whitespace between definitions. for a large test table this will quickly get out of hand.

Reply:
Fixed